packages/api-client/src/views/Request/RequestSection/RequestBody.vue

```diff
@@ -181,21 +181,52 @@ const updateRequestBody = (content: string) => {
 // }
 
 const updateActiveBody = (type: keyof typeof contentTypeOptions) => {
+  let contentTypeHeader = ''
   let bodyType: 'raw' | 'formData' | 'binary' = 'raw'
 
   if (type === 'multipartForm' || type === 'formUrlEncoded') {
     bodyType = 'formData'
+    contentTypeHeader = 'multipart/form-data'
   } else if (type === 'binaryFile') {
     bodyType = 'binary'
-  } else {
+    contentTypeHeader = 'application/octet-stream'
+  } else if (type !== 'none') {
     bodyType = 'raw'
+    const mappedType = type === 'other' ? 'html' : type
+    contentTypeHeader = `application/${mappedType}`
+    requestExampleMutators.edit(
+      activeExample.value.uid,
+      'body.raw.encoding',
+      mappedType,
+    )
   }
```

Comment:
This logic should really be made functional. Rather than a load of if/else statements and a lot of lets a function that takes the type and just returns `{ bodyType, contentTypeHeader }` would be clearer (and more testable)